core/src/main/java/org/apache/calcite/sql/type/SqlTypeName.java

```diff
@@ -810,6 +826,9 @@ public int getMinScale() {
     case INTERVAL_MINUTE:
     case INTERVAL_MINUTE_SECOND:
     case INTERVAL_SECOND:
+    case INTERVAL_MILLISECOND:
+    case INTERVAL_MICROSECOND:
+    case INTERVAL_NANOSECOND:
       return MIN_INTERVAL_FRACTIONAL_SECOND_PRECISION;
```

Comment:
Is this true? Don't these intervals have sub-second precision? What is this even being used for?

Reply:
BodoSQL and Snowflake doesn't support precision for any intervals. In calcite this attribute will be checked in validation step